src/nvim/log.c

```diff
@@ -32,6 +32,25 @@ static uv_mutex_t mutex;
 # include <execinfo.h>
 #endif
 
+#define LOCAL_TIME_STRING_LEN 20
+
+void get_local_time_string(char out[LOCAL_TIME_STRING_LEN])
```

Comment:
I don't think we use array parameters anywhere. They can be misleading. Better to just use a pointer.

Reply:
I was trying to avoid having `get_local_time_string()` allocate a `char[]` that needs to be freed by the caller. Is that what you had in mind? Or would you have the caller allocate the `char[]` and pass the length as a 2nd argument? (I don't have enough C experience to know what the "normal" approach would be here, I'm assuming it's one of those two?)

Reply:
While it's valid C syntax to declare the function as you did, it just gets collapsed down to a pointer.  Nothing enforces the length.

> I was trying to avoid having `get_local_time_string()` allocate a `char[]` that needs to be freed by the caller. Is that what you had in mind? Or would you have the caller allocate the `char[]` and pass the length as a 2nd argument?

It has to be one of the two. :) As long as it's clearly documented, either should be fine.  However, I would probably go with the latter.  Just make sure lengths are properly accounted for in `strftime()` and the fallback (`strlcpy()` or `strncpy()`).